lib/vsc/accounting/config/parser.py

```diff
@@ -0,0 +1,144 @@
+##
+# Copyright 2020-2020 Vrije Universiteit Brussel
+#
+# This file is part of vsc-accounting-brussel,
+# originally created by the HPC team of Vrij Universiteit Brussel (http://hpc.vub.be),
+# with support of Vrije Universiteit Brussel (http://www.vub.be),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/sisc-hpc/vsc-accounting-brussel
+#
+# vsc-accounting-brussel is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation v2.
+#
+# vsc-accounting-brussel is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with vsc-manage.  If not, see <http://www.gnu.org/licenses/>.
+#
+##
+"""
+Parser of configuration files for vsc.accounting
+
+@author: Alex Domingo (Vrije Universiteit Brussel)
+"""
+
+import os
+import appdirs
+import pkg_resources
+import configparser
+
+from vsc.utils import fancylogger
+from vsc.accounting.filetools import make_dir, copy_file
+from vsc.accounting.exit import error_exit
+
+
+class ConfigFile:
+    """
+    Parse contents of a config file in user's config dir
+    Generate default config file in user's config dir if it does not exist
+    """
+
+    def __init__(self, filename):
+        """
+        Set the location of the config file, create it if it does not exist and read its contents
+        - filename: (string) name of the config file
+        """
+        self.log = fancylogger.getLogger(name=self.__class__.__name__)
+
+        # Config file paths
+        self.usercfg = {
+            'name': filename,
+            'dir': appdirs.user_config_dir('vsc-accounting'),
```

Comment:
Handling of config files improved in https://github.com/sisc-hpc/vsc-accounting-brussel/pull/1/commits/04ae6a0bf25fc5dcff88cda33d72bb7bcccb8a10 and https://github.com/sisc-hpc/vsc-accounting-brussel/pull/1/commits/9f2947b8d3894bf9a49903aeb5a48d6ab3d213fd.

Since `accounting-report` will be executed by a user, config files in `~/.config` are still preferred. However, in lack of those, config files in `/etc` and `/etc/vsc-accountin` will also be considered. Additionally, I also added an option to be able to use a config file from any arbitrary path.